CONTRIBUTING.md

```diff
@@ -1,29 +1,81 @@
+- If you need help with your contribution, the Mixxx codebase or anything else related to Mixxx development don't hesitate to ask on the [development help stream](https://mixxx.zulipchat.com/#narrow/stream/247620-development-help).
+- The [development stream](https://mixxx.zulipchat.com/#narrow/stream/109171-development) is where day to day discussion about code, new features and releases happens.
+
+We also have lots of additional information for users and developers on the [Mixxx wiki](https://github.com/mixxxdj/mixxx/wiki), make sure to check it out if there's some piece of information you can't find.
+
+## How can I contribute? ##
+
+There are lots of ways you can contribute to Mixxx. Even if you don't know how to program, there's plenty of things you can do to help:
+
+ - [Report a bug](https://github.com/mixxxdj/mixxx/wiki/Reporting%20bugs)
+ - [Test the latest developments](https://github.com/mixxxdj/mixxx/wiki/Testing)
+ - [Translate Mixxx to other languages](https://github.com/mixxxdj/mixxx/wiki/Internationalization)
+ - [Improve the Mixxx manual](https://github.com/mixxxdj/manual#readme)
+ - [Improve the Mixxx website](https://github.com/mixxxdj/website)
+
+If you have some coding skills, or you are willing to learn them you can do even more:
+
+ - [Add a mapping for a new controller](https://github.com/mixxxdj/mixxx/wiki/Contributing%20mappings)
+ - [Fix bugs or program new features](#Your-first-Mixxx-code-contribution)
+
+We welcome programmers of all backgrounds and skill levels, and our community will do their best to help you with any problems you encounter.
+
+## Your first Mixxx code contribution ##
+
+### Compile Mixxx for the first time ###
+Your first goal is to get the Mixxx source code and compile it. The steps to do so depend on your operating system:
+- [Compiling on Linux](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20Linux)
+- [Compiling on Windows](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20Windows)
+- [Compiling on macOS](https://github.com/mixxxdj/mixxx/wiki/Compiling%20on%20macOS)
+
+### Pick an issue to work on ###
+You might already have in mind that feature you always dreamed of! If that's not the case or you want to start with something easy, you can ask for ideas on [Zulip](https://mixxx.zulipchat.com/#narrow/stream/247620-development-help) or pick an [easy issue](https://bugs.launchpad.net/mixxx/+bugs?field.tag=easy).
+
+### Time to code ###
+Now it's time to write some code!
+We recommend to [set up an IDE](https://github.com/mixxxdj/mixxx/wiki/Developer%20Tools) to assit you in the code writing process.
+
+We use [Git](https://git-scm.com/) for source control. Git is a distributed version control system that allows us to keep track of how our code changes and work together on it. If you are new to Git or source control in general, you can have a look at [Using Git](https://github.com/mixxxdj/mixxx/wiki/Using%20Git) to learn how.
```

Comment:
This can be moved up an shorten a bit.
The user most likely reads this file on GitHub.